packages/drivers/routerlicious-driver/src/restWrapper.ts

```diff
@@ -131,36 +138,64 @@ export class RouterliciousRestWrapper extends RestWrapper {
-		const fetchRequestConfig = axiosRequestConfigToFetchRequestConfig(translatedConfig);
+		const fetchRequestConfig = axiosBuildRequestInitConfig(translatedConfig);
 
 		const res = await this.rateLimiter.schedule(async () => {
 			const perfStart = performance.now();
-			const result = await fetch(...fetchRequestConfig).catch(async (error) => {
-				// Browser Fetch throws a TypeError on network error, `node-fetch` throws a FetchError
-				const isNetworkError = ["TypeError", "FetchError"].includes(error?.name);
-				const errorMessage = isNetworkError
-					? `NetworkError: ${error.message}`
-					: safeStringify(error);
-				// If a service is temporarily down or a browser resource limit is reached, RestWrapper will throw
-				// a network error with no status code (e.g. err:ERR_CONN_REFUSED or err:ERR_FAILED) and
-				// the error message will start with NetworkError as defined in restWrapper.ts
-				// If there exists a self-signed SSL certificates error, throw a NonRetryableError
-				// TODO: instead of relying on string matching, filter error based on the error code like we do for websocket connections
-				const err = errorMessage.includes("failed, reason: self signed certificate")
-					? new NonRetryableError(errorMessage, RouterliciousErrorTypes.sslCertError, {
-							driverVersion,
-						})
-					: new GenericNetworkError(errorMessage, errorMessage.startsWith("NetworkError"), {
-							driverVersion,
-						});
-				throw err;
-			});
+			const result = await fetch(completeRequestUrl, fetchRequestConfig).catch(
+				async (error) => {
+					// on failure, add the request entry into the retryCounter map to count the subsequent retries, if any
+					this.retryCounter.set(requestKey, requestRetryCount ? requestRetryCount + 1 : 1);
+
+					// Browser Fetch throws a TypeError on network error, `node-fetch` throws a FetchError
+					const isNetworkError = ["TypeError", "FetchError"].includes(error?.name);
+					const errorMessage = isNetworkError
+						? `NetworkError: ${error.message}`
+						: safeStringify(error);
+					// If a service is temporarily down or a browser resource limit is reached, RestWrapper will throw
+					// a network error with no status code (e.g. err:ERR_CONN_REFUSED or err:ERR_FAILED) and
+					// the error message will start with NetworkError as defined in restWrapper.ts
+					// If there exists a self-signed SSL certificates error, throw a NonRetryableError
+					// TODO: instead of relying on string matching, filter error based on the error code like we do for websocket connections
+					const err = errorMessage.includes("failed, reason: self signed certificate")
+						? new NonRetryableError(errorMessage, RouterliciousErrorTypes.sslCertError, {
+								driverVersion,
+								retryQueryParam: requestRetryCount,
+							})
+						: new GenericNetworkError(errorMessage, errorMessage.startsWith("NetworkError"), {
+								driverVersion,
+								retryQueryParam: requestRetryCount,
```

Comment:
nit: `retryCount` would be more accurate, right? Because we're tracking this value independently of the param